source/common/upstream/outlier_detection_impl.cc

```diff
@@ -152,5 +164,49 @@ void DetectorImpl::runCallbacks(HostPtr host) {
   }
 }
 
+void EventLoggerImpl::logEject(HostDescriptionPtr host, EjectionType type) {
+  // TODO: Log friendly host name (e.g., instance ID or DNS name).
+  // clang-format off
+  static const std::string json =
```

Comment:
minor, but having something like
std::string json = R"EOF(
  {
    
  }
  )EOF";
easier to read, but up to you.